src/canvas.c

```diff
@@ -466,6 +466,31 @@ wdTranslateWorld(WD_HCANVAS hCanvas, float dx, float dy)
     }
 }
 
+void
+wdSetWorldTransform(WD_HCANVAS hCanvas, const WD_MATRIX* pMatrix)
+{
+    if(d2d_enabled()) {
+        d2d_canvas_t* c = (d2d_canvas_t*) hCanvas;
+        dummy_D2D1_MATRIX_3X2_F m;
+
+        m._11 = pMatrix->m11;
+        m._12 = pMatrix->m12;
+        m._21 = pMatrix->m21;
+        m._22 = pMatrix->m22;
+        m._31 = pMatrix->dx;
+        m._32 = pMatrix->dy;
+        d2d_apply_transform(c, &m);
+    } else {
+        gdix_canvas_t* c = (gdix_canvas_t*) hCanvas;
+
+        dummy_GpMatrix* matrix;
+        gdix_vtable->fn_CreateMatrix2(pMatrix->m11, pMatrix->m12,
+                                      pMatrix->m21, pMatrix->m22,
+                                      pMatrix->dx, pMatrix->dy, &matrix);
+        gdix_vtable->fn_SetWorldTransform(c->graphics, matrix, dummy_MatrixOrderAppend);
```

Comment:
I am confused. Do you want `wdSetWorldTransform()` to 
1. just apply the provided transformation matrix, or 
2. should it combine it with the already set transformation matrix?

Both can make good sense, and I can imagine to have functions for both. But now it is some strange hybrid and it even makes something different for each back-end:
* The function name `wdSetWorldTransform()` imho indicates more the behavior 1.
* For D2D back-end, `d2d_apply_transform()` implements behavior 2.
* Fir GDI+ back-end, `GdipSetWorldTransform()` implements behavior 1, but your attempt to use the flag `dummy_MatrixOrderAppend` indicates that maybe you are attempting to implement behavior 2. If you want behavior 2, you should likely use `GdipMultiplyWorldTransform()` instead. 
   
Also note WinDrawLib also sometimes applies some transformation internally:
   * For D2D backend, there is the `0.5f, 0.5f` translation so both back-ends shared the coordinate system, `[0,0]` meaning left-top pixel.
   * With a RTL flag, the canvas automatically applies left-to-right mirroring transformation.

If `wdSetWorldTransform()` is supposed to implement behavior 1, it should still imho still keep this internal initial transformations.

Reply:
Ok, looking more into it it makes more sense to have a set instead of a multiply. This is because the interface is not a state machine as other graphic libraries use (with save/restore). However, the initial transform for rtl indeed should be kept.